napari/layers/shapes/shapes.py

```diff
@@ -55,6 +56,10 @@
 
 DEFAULT_COLOR_CYCLE = np.array([[1, 0, 1, 1], [0, 1, 0, 1]])
 
+# If there are more than this number of shapes, the thumnail won't update during
+# interactive events
+INTERACTIVE_THUMBNAIL_UPDATE_THRESHOLD = 100
```

Comment:
Something I wonder about is instead of having this threshold for interactive updating is to make all interactive updating off, AND have us use at most this number of randomly chosen shapes for a thumbnail (which is only ever 32x32 pixels), see what we do here in vectors https://github.com/napari/napari/blob/master/napari/layers/vectors/vectors.py#L676-L681

Reply:
Oh yeah, I forgot that we do that on some layers! That's a good idea. I agree that we should do the same here.

Reply:
Done!